### PR TITLE
[documenttranslation] Lazy loaded pageable for GetValues() and override WaitForCompletion implementation

### LINKED ISSUE #19502: How does `GetAllDocuments` differ from GetValues
- Should we have only GetValues? Is the name clear enough to indicate to the user what they are getting?
- If the user hasn't called `Update` and calls `GetValues` what should happen?
```
var operation = client.StartTranslation(...);
foreach (var document in operation.GetValues()) { }  -> ?
```

### PR DESCRIPTION
**Pull request by a-noyass**
Fixes #19502

### CODE REVIEW

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationOperation.cs [hunk 1]
```diff
@@ -445,22 +474,7 @@ public override AsyncPageable<DocumentStatusDetail> GetValuesAsync()
...
-                    return Page.FromValues(response.Value.Value, response.Value.NextLink, response.GetRawResponse());
-                }
-                catch (Exception)
-                {
-                    throw;
-                }
-            }
-
-            return PageableHelpers.CreateAsyncEnumerable(_ => Task.FromResult(_firstPage), NextPageFunc);
+            return GetAllDocumentsStatusAsync();
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationOperation.cs:477]: does this needs to be return await GetAllDocumentsStatusAsync() ?

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationOperation.cs [hunk 2]
```diff
@@ -185,10 +187,11 @@ internal DocumentTranslationOperation(DocumentTranslationRestClient serviceClien
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> used for the periodical service calls.</param>
         /// <returns>The last HTTP response received from the server.</returns>
         /// <remarks>
-        /// This method will periodically call UpdateStatusAsync till HasCompleted is true, then return the final result of the operation.
+        /// This method will periodically call UpdateStatusAsync till HasCompleted is true.
+        /// An API call is then made to retrieve the status of the documents.
         /// </remarks>
         public override ValueTask<Response<AsyncPageable<DocumentStatusDetail>>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
-            this.DefaultWaitForCompletionAsync(cancellationToken);
+            WaitForCompletionAsync(OperationHelpers.DefaultPollingInterval, cancellationToken);
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationOperation.cs:194]: Consider setting the polling interval to  `30s`  until we address https://github.com/Azure/azure-sdk-for-net/issues/19442

**Review by maririos — APPROVED** [review]
Could you link this PR to issue https://github.com/Azure/azure-sdk-for-net/issues/19502 ?

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationOperation.cs (hunk 1 above)
Comment (a-noyass) [reply sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationOperation.cs:477]: @maririos No the function is not actually async it just returns an AsyncPageable

Comment (maririos) [reply sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationOperation.cs:480]: duh. Right!

**Review by maririos — APPROVED** [review]

**Pull request merged**